contracts/starknet/src/escrow.cairo

```diff
@@ -243,18 +244,54 @@ mod Escrow {
         self.pausable.assert_not_paused();
         let eth_transfer_contract_felt: felt252 = self.eth_transfer_contract.read().into();
         assert(from_address == eth_transfer_contract_felt, 'Only PAYMENT_REGISTRY_CONTRACT');
-        assert(self.orders_pending.read(order_id), 'Order claimed or nonexistent');
 
-        let order = self.orders.read(order_id);
-        assert(order.recipient_address == recipient_address, 'recipient_address not match L1');
-        assert(order.amount == amount, 'amount not match L1');
+        _claim_payment(ref self, from_address, order_id, recipient_address, amount);
+    }
+
+    #[l1_handler]
+    fn claim_payment_batch(
+        ref self: ContractState,
+        from_address: felt252,
+        orders: Array<(u256, EthAddress, u256)>
+    ) {
+        let eth_transfer_contract_felt: felt252 = self.eth_transfer_contract.read().into();
+        assert(from_address == eth_transfer_contract_felt, 'Only PAYMENT_REGISTRY_CONTRACT');
+
+        let mut idx = 0;
+
+        loop {
+            if idx >= orders.len() {
+                break;
+            }
 
-        self.orders_pending.write(order_id, false);
-        let payment_amount = order.amount + order.fee;
+            let (order_id, recipient_address, amount) = *orders.at(idx);
 
-        IERC20Dispatcher { contract_address: self.native_token_eth_starknet.read() }
-            .transfer(self.mm_starknet_wallet.read(), payment_amount);
+            _claim_payment(ref self, from_address, order_id, recipient_address, amount);
```

Comment:
If one claim_payment fails, no claim will be made, is this the desired behaviour?

We should consider approving correct claims in L2, since big batches have higher chance one claim is wrong.

We could do this by replacing some `assert` statements with `if` statements

Reply:
We should make it atomic